r2-core/src/main/java/com/linkedin/r2/filter/transport/ClientRetryFilter.java

```diff
@@ -87,15 +87,7 @@ else if (ex instanceof StreamException)
         ex = new RetriableRequestException(retryAttr, ex);
       }
     }
-    else {
-      Throwable[] throwables = ExceptionUtils.getThrowables(ex);
-      for (Throwable throwable : throwables) {
-        if (throwable instanceof RetriableRequestException) {
-          ((RetriableRequestException) throwable).setDoNotRetryOverride(true);
-          break;
-        }
-      }
-    }
```

Comment:
Why was this implemented in the first place? what scenario are we trying to avoid'

Reply:
I originally thought client-side RetriableRequestException won't go through ClientRetryFilter. So as a safe-guard, I set do not retry override when the wire attribute is not present. But this is not necessary: 1. If we want to retry server-side exception, the wire attribute will be present and the exception will be wrapped into a retriable exception. 2. If we want to retry client-side exception, we should just pass the RetriableException to the next filter.

Reply:
makes sense, thanks for explaining